hera_mc/cm_redis_corr.py

```diff
@@ -72,25 +36,28 @@ def cminfo_redis_snap(cminfo, redis_info=None):
     snap_to_ant = {}
     ant_to_snap = {}
     all_snap_inputs = {}
-    for antn, ant in enumerate(cminfo['antenna_numbers']):
-        name = cminfo['antenna_names'][antn]
-        pol_info = {}
-        for pol_snap in cminfo['correlator_inputs'][antn]:
-            pol = pol_snap.lower()[0]
-            if pol in ['e', 'n'] and '>' in pol_snap:
-                pol_info[pol] = pol_snap
+    snap_to_serial = {}
+    for _n, ant in enumerate(cminfo['antenna_numbers']):
+        name = cminfo['antenna_names'][_n]
+        corr_inp = cminfo['correlator_inputs'][_n]
+        snap_snr = cminfo['snap_serial_numbers'][_n]
         ant_to_snap[ant] = {}
-        for pol, psnap in pol_info.items():
-            snapi, channel = snap_part_to_host_input(psnap, redis_info=redis_info)
-            ant_to_snap[ant][pol] = {'host': snapi, 'channel': channel}
-            snap_to_ant.setdefault(snapi, [None] * 6)
-            snap_to_ant[snapi][channel] = name + pol.upper()
-            snap, adc_num = snap_part_to_host_input(psnap, None)
-            all_snap_inputs.setdefault(snap, [])
-            all_snap_inputs[snap].append(adc_num)
+        for _i, psnap in enumerate(corr_inp):
+            pol = psnap.lower()[0]
+            if pol not in ['e', 'n'] or '>' not in psnap:
+                warnings.warn(f"{psnap} is not an allowed correlator input")
+                continue
+            snap_input, snap_hostname = psnap.split('>')
+            channel = int(snap_input[1:]) // 2  # divide by 2 because ADC is in demux 2
+            ant_to_snap[ant][pol] = {'host': snap_hostname, 'channel': channel}
+            snap_to_ant.setdefault(snap_hostname, [None] * 6)
+            snap_to_ant[snap_hostname][channel] = name + pol.upper()
+            snap_to_serial[snap_hostname] = snap_snr[_i]  # This can/does overwrite per pol
```

Comment:
Rather than unconditionally overwriting this value for each polarization, it might be better to check that they're the same. Maybe something like this?
```python
try:
    existing_hostname = snap_to_serial[snap_hostname]
    if existing_hostname != snap_snr[_i]:
        raise ValueError(f"The snap_hostname {snap_hostname} is different between polarizations")
except KeyError:
    # key does not yet exist in dict; add it
    snap_to_serial[snap_hostname] = snap_snr[_i]
```

Reply:
I added the suggested check.  However, I issue a warning not raise an error since I can imagine at some point we'd want to run some tests where they aren't the same snap and we want the system to still work in that case.  I kept the warning since we do want to be alerted if that does happen (purposely or inadvertently)

Reply:
I now correctly raise an error if the hostname to serial numbers are not consistent.